examples/text_labels_and_annotations/mathtext_fontset_example.py

```diff
@@ -0,0 +1,77 @@
+"""
+========================
+Mathtext Fontset Example
+========================
+
+"""
+
+import matplotlib.pyplot as plt
+
+plt.figure(figsize=(20, 10))
+
+"""A simple plot for the background"""
+plt.plot(range(11), color="0.9", label=r'Math Font Example')
+
+
+"""
+Here we hold the text, we will mix normal text and math text
+(math text is enclosed by *$*).
+"""
+msg = r"This is a normal text and "
+msg+= r"$this\ is\ a\ text\ in\ math\ mode,\ look,\ I\ can\ do\ this\ "
+msg+= r"\int_{0}^{\infty } x^2 dx !$"
+
+"""
+Now we set the text to be shown, its (x, y) coordinates, it's size
+and the family of fonts for the math text.
+"""
+plt.text(0.2, 6, msg, size=24, mathtext_fontset='dejavusans')
+
+"""
+The parameter *mathtext_fontset* migth be too long for some, so we
+aliased it to *math_font*, the following code will work just as well. 
+"""
+plt.text(0.2, 4, msg, size=24, math_font='dejavuserif')
+
+"""
+Also, we can call *mathtext_fontset* and *math_font* in most places where 
+there is text, like in the title:
+"""
+
+titleMsg = r"$This\ is\ a\ title\ in\ math\ mode,\ look:\ "
+titleMsg+= r"\int_{0}^{\infty } x^2 dx !$"
+
+plt.title(titleMsg, math_font='cm', size=24)
+
+"""
+The currently supported fonts for text in math mode are available in a file
+called *matplotlibrc.template*, see the :ref:`matplotlib rc file
+<customizing-with-matplotlibrc-files>` for details.
+"""
+
+"""
+Note how the normal text is not changed by the *math_font* parameter.
+"""
+plt.show()
+
+"""
+Before *math_font* was implemented, the only way to change the font in math 
+mode was by using *rcParams['mathtext.fontset']*, but it would only set one 
+font for the entire plot. The following code will only show math texts in 
+the "cm" font (because it was the last one set before calling *plt.show()*).
+"""
+
+plt.figure(figsize=(20, 10))
+plt.plot(range(11), color="0.9", label=r'Math Font Example with rcParams')
+
+plt.rcParams['mathtext.fontset']='dejavusans' # will be overriden
+plt.text(0.2, 6, msg, size=24)
+
+plt.rcParams['mathtext.fontset']='dejavuserif' # will be overriden
+plt.text(0.2, 4, msg, size=24)
+
+plt.rcParams['mathtext.fontset']='cm' # Last math font set, all math text
+                                      # will be rendered using this font
+plt.title(titleMsg, size=24)
+
+plt.show()
```

Comment:
Please don't show negative examples. They bear the risk of people blindly copying code without reading the explanations, and then they wonder why it does not work.

IMHO it's sufficient to leave this out and instead add the following text at the top.

~~~
The global setting :rc:`mathtext.fontset` is applied when rendering the figure,
so that all texts in the figure use the same math font. Using the keyword
argument `mathtext_fontset` allows changing the math font of individual texts.
~~~